pkg/operator/resource/resourceapply/admissions.go

```diff
@@ -2,76 +2,159 @@ package resourceapply
+// it will be created. If it does exist, the metadata of the required
+// validatingwebhookconfiguration will be merged with the existing validatingwebhookconfiguration
+// and an update performed if the validatingwebhookconfiguration spec and metadata differ from
+// the previously required spec and metadata based on generation change.
+//
+// expectedGeneration set to 0 disables generation tracking, forcing redeployment logic.
+func ApplyValidatingWebhookConfiguration(client admissionregistrationclientv1.ValidatingWebhookConfigurationsGetter, recorder events.Recorder,
+	requiredOriginal *admissionregistrationv1.ValidatingWebhookConfiguration, expectedGeneration int64) (*admissionregistrationv1.ValidatingWebhookConfiguration, bool, error) {
+	if requiredOriginal == nil {
+		return nil, false, fmt.Errorf("Unexpected nil instead of an object")
+	}
+	required := requiredOriginal.DeepCopy()
+
+	existing, err := client.ValidatingWebhookConfigurations().Get(context.TODO(), required.GetName(), metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
-		actual, err := client.MutatingWebhookConfigurations().
-			Create(context.TODO(), required, metav1.CreateOptions{})
+		actual, err := client.ValidatingWebhookConfigurations().Create(context.TODO(), required, metav1.CreateOptions{})
 		reportCreateEvent(recorder, required, err)
-		return actual, true, err
-	}
-	if err != nil {
+		if err != nil {
+			return nil, false, err
+		}
+		return actual, true, nil
+	} else if err != nil {
 		return nil, false, err
 	}
 
 	modified := resourcemerge.BoolPtr(false)
 	existingCopy := existing.DeepCopy()
 
 	resourcemerge.EnsureObjectMeta(modified, &existingCopy.ObjectMeta, required.ObjectMeta)
-	contentSame := equality.Semantic.DeepEqual(existingCopy.Webhooks, required.Webhooks)
-	if contentSame && !*modified {
+	if !*modified && existingCopy.GetGeneration() == expectedGeneration {
 		return existingCopy, false, nil
 	}
-
-	if klog.V(4).Enabled() {
-		klog.Infof("ValidatingWebhookConfiguration %q changes: %v", required.Name, JSONPatchNoError(existing, required))
+	// at this point we know that we're going to perform a write.  We're just trying to get the object correct
+	toWrite := existingCopy // shallow copy so the code reads easier
+	// Providing upgrade compatibility with service-ca-bundle operator
+	// and ignore clientConfig.caBundle changes on "inject-cabundle" label
+	if required.GetAnnotations() != nil && required.GetAnnotations()[genericCABundleInjectorAnnotation] != "" {
+		// Populate unpopulated webhooks[].clientConfig.caBundle fields from existing resource
+		copyValidatingWebhookCABundle(existing, required)
 	}
+	toWrite.Webhooks = required.Webhooks
 
-	actual, err := client.MutatingWebhookConfigurations().Update(context.TODO(), required, metav1.UpdateOptions{})
+	klog.V(4).Infof("ValidatingWebhookConfiguration %q changes: %v", required.GetNamespace()+"/"+required.GetName(), JSONPatchNoError(existing, toWrite))
```

Comment:
there was a reason for `if klog.V(4).Enabled() {`

Reply:
It should be the same, according to docs: 
```golang
// Thus, one may write either
//	if glog.V(2).Enabled() { klog.Info("log this") }
// or
//	klog.V(2).Info("log this")
// The second form is shorter but the first is cheaper if logging is off because it does
// not evaluate its arguments.
```